elisa/src/org/labkey/elisa/ElisaSampleFilePropertyHelper.java

```diff
@@ -0,0 +1,119 @@
+    {
+        if (_sampleProperties != null)
+            return _sampleProperties;
+
+        File metadataFile = getSampleMetadata(request);
+        if (metadataFile == null)
+            throw new ExperimentException("No metadata or data file provided");
+
+        Map<String, Map<DomainProperty, String>> allProperties = new HashMap<>();
+        try
+        {
+            DataLoaderFactory factory = DataLoaderService.get().findFactory(metadataFile, null);
+            DataLoader loader = factory.createLoader(metadataFile, true);
+
+            String plateColumnName = "Plate Name";
+            String sampleColumnName = "Sample";
+            String wellLocationColumnName = "Well";
+            String spotColumnName = "Spot";
+
+            boolean hasPlateColumn = Arrays.stream(loader.getColumns()).anyMatch(c -> c.getColumnName().equalsIgnoreCase(plateColumnName));
+            boolean hasSampleColumn = Arrays.stream(loader.getColumns()).anyMatch(c -> c.getColumnName().equalsIgnoreCase(sampleColumnName));
+            boolean hasWellLocationColumn = Arrays.stream(loader.getColumns()).anyMatch(c -> c.getColumnName().equalsIgnoreCase(wellLocationColumnName));
+            boolean hasSpotColumn = Arrays.stream(loader.getColumns()).anyMatch(c -> c.getColumnName().equalsIgnoreCase(spotColumnName));
+
+            if (!hasPlateColumn)
+                throw new ExperimentException("Sample metadata file does not contain required column \"" + plateColumnName + "\".");
+            if (!hasSampleColumn)
+                throw new ExperimentException("Sample metadata file does not contain required column \"" + sampleColumnName + "\".");
+            if (!hasWellLocationColumn)
+                throw new ExperimentException("Sample metadata file does not contain required column \"" + wellLocationColumnName + "\".");
+            if (!hasSpotColumn)
+                throw new ExperimentException("Sample metadata file does not contain required column \"" + spotColumnName + "\".");
+
+            for (Map<String, Object> row : loader)
+            {
+                String wellLocation = String.valueOf(row.get(wellLocationColumnName));
+                String plateName = String.valueOf(row.get(plateColumnName));
+                Integer spot = (Integer)row.get(spotColumnName);
+
+                if (wellLocation != null && plateName != null)
+                {
+                    String sampleWellGroup = getSampleWellGroupFromLocation(plateName, spot, wellLocation);
+                    if (sampleWellGroup != null)
+                    {
+                        Map<DomainProperty, String> sampleProperties = allProperties.computeIfAbsent(sampleWellGroup, k -> new HashMap<>());
+                        for (DomainProperty property : _domainProperties)
+                        {
+                            Object value = getValue(row, property);
+                            sampleProperties.put(property, value != null ? value.toString() : null);
```

Comment:
we are putting prop/values into `sampleProperties` here but then below we set `_sampleProperties = allProperties`, just checking to make sure this wasn't supposed to be a put into `allProperties`.